Salehi2025/data.R

```diff
@@ -17,237 +17,127 @@ library(glmnet)
-adult_metadata$shannon_diversity <- diversity(t(assay_data_clean), index = "shannon")
-
+# Subset TSE to retain only samples present in filtered metadata
+tse_class_top5 <- tse_class_top5[, colnames(tse_class_top5) %in% adult_metadata$acc]
+colData(tse_class_top5) <- DataFrame(adult_metadata)
 
-# This section creates dummy variables (one-hot encoding) for categorical 
-# predictors to be used in the GLM and Bayesian regression models
+# Remove samples with zero total relative abundance across all features
+tse_class_top5 <- tse_class_top5[, colSums(assay(tse_class_top5, "relabundance")) > 0]
 
-# Copy metadata to a temporary dataframe
-temp_df <- adult_metadata
-
-# Set the correct factor level order for age categories
-temp_df$age_category <- factor(temp_df$age_category,
-                               levels = c("Middle-Aged Adult", "Infant",
-                                          "Toddler", "Children", "Teenager",
-                                          "Young Adult", "Older Adult", "Oldest Adult"))
-
-# Create binary variable: High income (1) vs. other (0)
-temp_df$income_group_HIC <- ifelse(temp_df$World_Bank_Income_Group == "High income", 1, 0)
-
-# Create binary variable for antibiotic usage: high (1) vs. low (0)
-temp_df$Usage_high <- ifelse(temp_df$Usage < 10, 0, 1)
+# Calculate Shannon diversity index directly using mia's built-in support
+tse_class_top5 <- addAlpha(
+  tse_class_top5,
+  assay.type = "relabundance",
+  index = "shannon",
+  name = "shannon_diversity"
+)
 
-# Convert gender to binary numeric variable: Men = 0, Women = 1
-temp_df$sex_num_Men <- ifelse(temp_df$gender == "Men", 0, 1)
+# Update adult_metadata from the modified TSE object
+adult_metadata <- as.data.frame(colData(tse_class_top5))
 
-# Perform dummy (one-hot) encoding for selected predictors
-temp_df_one_hot <- model.matrix(~ 0 + 
-                                  sex_num_Men +
-                                  continent + 
-                                  age_category + 
-                                  income_group_HIC + 
-                                  Usage_high,
-                                data = temp_df) %>%
-  as.data.frame()
+# Set age_category as a factor in natural age order
+adult_metadata$age_category <- factor(adult_metadata$age_category,
+                                      levels = c("Infant", "Toddler", "Children", "Teenager", 
+                                                 "Young Adult", "Middle-Aged Adult", "Older Adult", "Oldest Adult")
+)
 
-# Clean column names by replacing spaces and dashes with underscores
-colnames(temp_df_one_hot) <- gsub(" |-", "_", colnames(temp_df_one_hot))
+# Create binary and factor variables used in modeling
+adult_metadata$income_group_HIC <- ifelse(adult_metadata$World_Bank_Income_Group == "High income", 1, 0)
+adult_metadata$Usage_high <- ifelse(adult_metadata$Usage < 10, 0, 1)
+adult_metadata$sex_num_Men <- ifelse(adult_metadata$gender == "Men", 0, 1)

```

Comment:
If I am correct then isnt this now like "Men"=0, not "Men" = 1?

This seems counterintuitive if the variable name is sex_num_Men? Shouldnt such variable be like Men=1? not men = 0?

btw in general I recommend not to mix lowercase and uppercase in these internal (not print ready) variable names, it is a potential source of confusion.